com.unity.multiplayer.mlapi/Runtime/Logging/FileLogger.cs

```diff
@@ -0,0 +1,38 @@
+using System.IO;
+
+namespace MLAPI.Logging
+{
+    /// <summary>
+    /// Helper class for logging
+    /// Saves to a file identified by the process ID of each client
+    /// This is useful when debugging what happens on multiple machines
+    /// ! Only recommended for debugging, not for production !
+    /// </summary>
+    public class FileLogger
+    {
+        public static FileLogger Get() { return instance; }
+        private static FileLogger instance = new FileLogger();
+        private StreamWriter writer;
+
+        private FileLogger()
+        {
+            System.Diagnostics.Process p = System.Diagnostics.Process.GetCurrentProcess();
+            writer = new StreamWriter("log." + p.Id + ".txt");
+            writer.AutoFlush = true;
+        }
+
+        ~FileLogger()
+        {
+            writer.Close();
+        }
```

Comment:
instead of this, you should implement `IDisposable` with `Dispose()` method instead and put `writer.Dispose()` in it.
I also believe you could set [`AutoFlush`](https://docs.microsoft.com/en-us/dotnet/api/system.io.streamwriter.autoflush) to `false` and expose [`Flush()`](https://docs.microsoft.com/en-us/dotnet/api/system.io.streamwriter.flush) API.

Reply:
I agree with the dispose. 
But not with the AutoFlush. Since this is debug functionality, you want the info to get out asap. You might be stopped on a breakpoint soon after, or get other crash/exceptions. Auto-flushing comes at very little cost and provides peace-of-mind that you'll get your logs.